### PR TITLE
fix extra datum with simple ada transfer

### LINKED ISSUE #340: Extra datum with simple Ada transfer
Attempting to submit the transaction in `examples/Pkh2Pkh.purs` results in:

```
{
  "type": "jsonwsp/response",
  "version": "1.0",
  "servicename": "ogmios",
  "methodname": "SubmitTx",
  "result": {
    "SubmitFail": [
      {
        "extraDataMismatch": {
          "provided": "6c5f384a73bacc120b6b78c55d1078d0e422898b75b5434127ba028479747347",
          "inferredFromParameters": null
        }
      }
    ]
  },
  "reflection": { "id": "SubmitTx-l29wf6qu", "step": "INIT" }
}
```
Avoiding `finalizeTx` entirely solves this, but that is not a good solution

### PR DESCRIPTION
**Pull request by Luis-omega**
fixes #340

### CODE REVIEW

File: server/src/Api/Handlers.hs [hunk 1]
```diff
@@ -102,7 +103,14 @@ finalizeTx (FinalizeRequest {tx, datums, redeemers}) = do
   decodedDatums <-
     throwDecodeErrorWithMessage "Failed to decode datums" $
       traverse decodeCborDatum datums
-  let languages = Set.fromList [PlutusV1]
+  let witnessScripts = (Tx.txscripts' . Tx.wits) decodedTx
+      (Value.Value ada maps) = (Tx.mint . Tx.body) decodedTx
+      languages =
+        if Map.null witnessScripts
+          && Map.null maps
+          && ada == 0
```
Comment (klntsky) [inline server/src/Api/Handlers.hs:111]: This looks a bit different from a similar change in [another PR](https://github.com/Plutonomicon/cardano-transaction-lib/pull/395/files#diff-4f1478ccd6d4b99631ab8cf039548961315e8091fe4b5dafa1d6561a38bc670eR106).

What documentation this code is based on?

Comment (ngua) [reply server/src/Api/Handlers.hs:111]: @klntsky These things are often undocumented and the changes from the other PR were a related but separate issue

Comment (ngua) [inline server/src/Api/Handlers.hs:111]: I don't see why the Ada value would have an effect here

File: server/src/Api/Handlers.hs [hunk 2]
```diff
@@ -102,7 +103,14 @@ finalizeTx (FinalizeRequest {tx, datums, redeemers}) = do
   decodedDatums <-
     throwDecodeErrorWithMessage "Failed to decode datums" $
       traverse decodeCborDatum datums
-  let languages = Set.fromList [PlutusV1]
+  let witnessScripts = (Tx.txscripts' . Tx.wits) decodedTx
+      (Value.Value ada maps) = (Tx.mint . Tx.body) decodedTx
```
Comment (ngua) [inline server/src/Api/Handlers.hs:107]: I think `assets` might be a clearer name than `maps`

**Review by ngua — COMMENTED** [review]
I'll merge this with the changes from #395 and then we can merge this. I don't think we need to bother with checking the Ada value in `mint` since only minting scripts should affect this

**Review by ngua — APPROVED** [review]

**Pull request merged**